### PR TITLE
Initial infrastructure for v1.0.0 release

### PR DESCRIPTION
**Pull request by ed-xmos**
Note this is the first of many PRs. I'm getting the basics in place first with this one.

https://xmosjira.atlassian.net/browse/UA-331
https://xmosjira.atlassian.net/browse/UA-333
https://xmosjira.atlassian.net/browse/UA-382

Docs are covered by #10

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+                                        xcoreLibraryChecks("${REPO}", false)
+                                        // junit "junit_lib.xml"
+                                    } // withEnv
+                                } // withTools
+                            } // Venv
+                        } // dir
+                    } // steps
+                }
+                stage('Docs') {
+                    environment { XMOSDOC_VERSION = "v4.0" }
```
Comment (danielpieczko) [inline Jenkinsfile:65]: There is a newer xmosdoc version v5.something

File: examples/app_xu316_i2s_loopback/README.rst [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+This example contains a simple application and demonstrates how to use this library to abstract away the board hardware setup code.
+
+The application supports a simple I2S digital loopback which plays the samples received on all 8 ADC channels back out on the 8 DAC channels.
+All of the I2S configuration is set in the defines at the top of ``main.xc`` which allows different sample rates, master clock frequencies and number of data bits etc. to be set before compilation. Sample rates of 44.1 kHz to 192 kHz have been tested.
+
+Two build configurations are included:
+
+    - XMOS_MASTER - This configures the ADCs and DACs to I2S slave and the xcore.ai device drives the I2S clocks. The on-chip application PLL is configured to drive the master clock to the mixed signal devices.
+    - XMOS_SLAVES - This configures one the DACs to I2S master and the remaining DAC, all ADCs and the xcore.ai device to I2S slave. The on-chip application PLL is configured to drive the master clock to the mixed signal devices.
```
Comment (danielpieczko) [inline examples/app_xu316_i2s_loopback/README.rst:18]: Do you need vertical bars on all the "I2S" instances to get the substitution?

File: examples/app_xu316_i2s_loopback/README.rst [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+  * lib_i2c (www.github.com/xmos/lib_i2c)
+  * lib_i2s (www.github.com/xmos/lib_i2s)
+  * lib_sw_pll (www.github.com/xmos/lib_sw_pll)
+  * lib_xassert (www.github.com/xmos/lib_xassert)
+
+
+Required hardware
+.................
+
+The hardware targeted is the `XK-AUDIO-XU316-MC board <hhttps://www.xmos.com/download/XCORE_AI-Multichannel-Audio-Platform-1V1-Hardware-Manual(1V1).pdf>`_ although thanks to this library, porting it to other platforms supported by ``lib_board_support`` should be trivial.
```
Comment (danielpieczko) [inline examples/app_xu316_i2s_loopback/README.rst:54]:
```suggestion
The hardware targeted is the `XK-AUDIO-316-MC board <hhttps://www.xmos.com/download/XCORE_AI-Multichannel-Audio-Platform-1V1-Hardware-Manual(1V1).pdf>`_ although thanks to this library, porting it to other platforms supported by ``lib_board_support`` should be trivial.
```

**Review by danielpieczko — APPROVED** [review]
A couple of minor comments, but otherwise looks good.

File: examples/app_xu316_i2s_loopback/README.rst (hunk 2 above)
Comment (ed-xmos) [reply examples/app_xu316_i2s_loopback/README.rst:18]: Oh yes. Good point

**Pull request merged**